### PR TITLE
[TM] Add specs for ModalManager

### PR DESCRIPTION
**Pull request by abhiin1947**
## Summary
Part of #24875

## Changelog
[General] [Added] - Add TurboModule spec for ModalManager

## Test Plan
Flow and tests pass

### CODE REVIEW

File: Libraries/Modal/NativeModalManager.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+'use strict';
+
+import type {TurboModule} from 'RCTExport';
+import * as TurboModuleRegistry from 'TurboModuleRegistry';
+
+export interface Spec extends TurboModule {
+  // RCTEventEmitter
+  +addListener: (eventName: string, handler: Function) => Object;
```
Comment (ericlewis) [inline Libraries/Modal/NativeModalManager.js]: remove these listeners, since it is not an event emitter

File: Libraries/Modal/Modal.js [hunk 2]
```diff
@@ -23,9 +21,7 @@ const View = require('../Components/View/View');
 const RCTModalHostView = require('./RCTModalHostViewNativeComponent');
 
 const ModalEventEmitter =
-  Platform.OS === 'ios' && NativeModules.ModalManager
-    ? new NativeEventEmitter(NativeModules.ModalManager)
-    : null;
+  Platform.OS === 'ios' ? require('./NativeModalManager').default : null;
```
Comment (ericlewis) [inline Libraries/Modal/Modal.js]: change this to an import, also, do the condition in `NativeModalManager.js` not here.

File: Libraries/Modal/NativeModalManager.js (hunk 1 above)
Comment (fkgozali) [reply Libraries/Modal/NativeModalManager.js]: RCTModalManager actually extends RCTEventEmitter

Comment (abhiin1947) [reply Libraries/Modal/NativeModalManager.js]: What @fkgozali said I couldn't get flow to run otherwise.

**Pull request closed without merging**